lib/airbrake-ruby/filters/keys_filter.rb

```diff
@@ -55,7 +55,9 @@ def filter_hash(hash)
 
       def filter_url_params(url)
         url.query = Hash[URI.decode_www_form(url.query)].map do |key, val|
-          should_filter?(key) ? "#{key}=[Filtered]" : "#{key}=#{val}"
+          encoded_key = URI.encode_www_form_component(key)
+          encoded_val = URI.encode_www_form_component(val)
+          should_filter?(key) ? "#{encoded_key}=[Filtered]" : "#{encoded_key}=#{encoded_val}"
```

Comment:
Looks like the build is failing because of line length. Could you convert this into a simple `if` statement and avoid temporary variables, so this code can be a little bit faster? Given that this code is needed to support older Rubies, could you add a brief comment describing why we need to encode components?